fast_api/routes/heuristic.py

```diff
@@ -0,0 +1,295 @@
+        ),
+        "sourceStatistics",
+    )
+    if statistics is not None:
+        data["sourceStatistics"] = statistics["data"]["sourceStatistics"]
+    return pack_json_result({"data": {"informationSourceBySourceId": data}})
+
+
+@router.get(
+    "/{project_id}/{payload_id}/payload-by-id",
+    dependencies=[Depends(auth_manager.check_project_access_dep)],
+)
+def get_payload_by_payload_id(
+    project_id: str,
+    payload_id: str,
+):
+    data = sql_alchemy_to_dict(get_payload_with_heuristic_type(project_id, payload_id))
+    return pack_json_result({"data": {"payloadByPayloadId": data}})
+
+
+@router.get(
+    "/{project_id}/{heuristic_id}/lf-on-10-records",
+    dependencies=[Depends(auth_manager.check_project_access_dep)],
+)
+def get_labeling_function_on_10_records(
+    project_id: str,
+    heuristic_id: str,
+):
+    data = sql_alchemy_to_dict(
+        payload_manager.get_labeling_function_on_10_records(project_id, heuristic_id)
+    )
+    records = []
+    for record in data.records:
+        records.append(
+            {
+                "recordId": record.record_id,
+                "calculatedLabels": record.calculated_labels,
+                "fullRecordData": record.full_record_data,
+            }
+        )
+    final_data = {
+        "records": records,
+        "containerLogs": data.container_logs,
+        "codeHasErrors": data.code_has_errors,
+    }
+    return {"data": {"getLabelingFunctionOn10Records": final_data}}
+
+
+@router.get(
+    "/{project_id}/model-callbacks-overview-data",
+    dependencies=[Depends(auth_manager.check_project_access_dep)],
+)
+def get_model_callbacks_overview_data(
+    request: Request,
+    project_id: str,
+):
+    data = manager.get_overview_data(project_id, is_model_callback=True)
+    return pack_json_result({"data": {"modelCallbacksOverviewData": data}})
+
+
+@router.get(
+    "/{project_id}/access-link",
+    dependencies=[Depends(auth_manager.check_project_access_dep)],
+)
+def get_access_link(
+    request: Request,
+    project_id: str,
+    link_id: str,
+):
+    accessLink = access_link_manager.get(link_id)
+
+    data = {
+        "id": str(accessLink.id),
+        "link": accessLink.link,
+        "isLocked": accessLink.is_locked,
+    }
+
+    return pack_json_result({"data": {"accessLink": data}})
+
+
+@router.post(
+    "/{project_id}/{information_source_id}/toggle-heuristic",
+    dependencies=[Depends(auth_manager.check_project_access_dep)],
+)
+def toggle_heuristic(
+    request: Request,
+    project_id: str,
+    information_source_id: str,
+):
+    manager.toggle_information_source(project_id, information_source_id)
+    notification.send_organization_update(
+        project_id, f"information_source_updated:{information_source_id}"
+    )
+    return pack_json_result({"data": {"toggleInformationSource": {"ok": True}}})
+
+
+@router.post(
+    "/{project_id}", dependencies=[Depends(auth_manager.check_project_access_dep)]
+)
```

Comment:
i think this should be renamed, the route has only projectId but what it actually does is selecting or deselecting all heuristics